### ISSUE TITLE
parse_datetime wrongly accepts datetimes that mix extended and basic format

### ISSUE THREAD

**Issue by ExplodingCabbage** [closeio/ciso8601#71]
This isn't legal under ISO 8601 and should throw an error:

```
>>> ciso8601.parse_datetime('20010203T04:05:06Z')
datetime.datetime(2001, 2, 3, 4, 5, 6, tzinfo=datetime.timezone.utc)
```

So should this:

```
>>> ciso8601.parse_datetime('20010203T04:05')
datetime.datetime(2001, 2, 3, 4, 5)
```

The first is forbidden by the combination of section 4.3.2, which specifies the format for "complete" date and time representations, and section 4.4.4, about date *interval* representations. Section 4.3.2 says:

> **The  hyphen  [-] and  the  colon  [:]  shall  be  used, in accordance with 4.4.4**, as separators within the date and time of day expressions, respectively, when required.

And section 4.4.4.1, says:

> When the application identifies the need for a complete representation of a time interval, identified by its start and its end, it  shall  use an expression in accordance with  4.4.2  combining any  two  complete date and time of day representations as defined in 4.3.2, **provided that the resulting expression is either consistently in basic format or consistently in extended format**.

It's kind of weird that the section about how to represent complete datetimes basically says "Hey, to know all the rules about how to do this, go forward in the spec and read the unrelated section about datetime intervals, because we actually put some of our rules about what a legal datetime representation in *there* for some reason", but as far as I can tell that's exactly what section 4.3.2 says. It would've been more natural to put *all* the rules about what a legal complete datetime representation is in the section about complete datetime representations and refer back to *that* from the section about intervals, but that's not what the authors did. Meh.

The second example I gave above, with a "reduced accuracy" datetime, is more straightforward; it's banned by section 4.3.3, which says

> For reduced accuracy, decimal or expanded representations of date and time of day, any of the
representations in *[bla]* followed immediately by the time designator [T] may be combined with any of the representations in *[bla]* provided that
>
> a) *[bla]*;
> b) *[bla]*;
> c) *[bla]*;
> d) **the expression shall either be completely in basic format**, in which case the minimum number of separators necessary for the required expression is used, **or completely in extended format**, in which case additional separators shall be used in accordance with 4.1 and 4.2.

**Comment by movermeyer** [comment]
Yes. Good catch. :smile: 

I like [Wikipedia's description](https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations) for its clarity:

> Either basic or extended formats may be used, but both date and time must use the same format.

Fixing this shouldn't be to difficult of a change. I'd welcome a PR for this if you're up for it. Otherwise, I'll get this fixed as I get the time.

**Comment by movermeyer** [comment]
I'll start fixing this.

**Comment by movermeyer** [comment]
Should be fixed in PR #73.
